### PR TITLE
Skeleton code for Reservation model, dao, service, controller

### PR DESCRIPTION
**Pull request by kebrahim**
Issue #17 

- [x] Tests pass
- [X] Appropriate changes to README are included in PR

### CODE REVIEW

File: callmeback/src/main/java/org/google/callmeback/api/ReservationController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("api/v1/reservation")
```
Comment (mscharfstein) [inline callmeback/src/main/java/org/google/callmeback/api/ReservationController.java:19]: in the docs we have reservations (plural) and i think that is the standard in REST APIs, so maybe update?

Comment (egracer) [reply callmeback/src/main/java/org/google/callmeback/api/ReservationController.java:19]: +1

Comment (kebrahim) [reply callmeback/src/main/java/org/google/callmeback/api/ReservationController.java]: done

File: callmeback/src/main/java/org/google/callmeback/api/ReservationController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  @GetMapping
+  public List<Reservation> getAllReservations() {
+    return reservationService.getAllReservations();
+  }
+
+  @GetMapping(path = "{id}")
+  public Reservation getReservationById(@PathVariable("id") UUID id) {
+    return reservationService.getReservationById(id).orElse(null);
```
Comment (egracer) [inline callmeback/src/main/java/org/google/callmeback/api/ReservationController.java:42]: Do we want to return an empty response with a 200 status code, or a 404? I'm not sure how Spring Boot converts a null response to HTTP status codes.

File: callmeback/src/main/java/org/google/callmeback/dao/FakeReservationDataAccessService.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.google.callmeback.dao.ReservationDao;
+import org.google.callmeback.model.Reservation;
+import org.springframework.stereotype.Repository;
+
+@Repository("fakeReservationDao")
+public class FakeReservationDataAccessService implements ReservationDao {
+
+  private static List<Reservation> DB =
+      Arrays.asList(new Reservation(UUID.randomUUID(), "Jason Mendoza"));
```
Comment (egracer) [inline callmeback/src/main/java/org/google/callmeback/dao/FakeReservationDataAccessService.java:17]: Good Place FTW 👍

File: callmeback/src/main/java/org/google/callmeback/api/ReservationController.java (hunk 2 above)
Comment (kebrahim) [reply callmeback/src/main/java/org/google/callmeback/api/ReservationController.java:42]: it returns a 200 for now (i'll leave that in a comment). i'm going to leave this as-is & merge & then we can discuss as a group how we want to handle these types of requests.

**Pull request merged**